src/hooks/useLoadDisciplines.ts

```diff
@@ -1,37 +1,35 @@
 import { Discipline, ENDPOINTS } from '../constants/endpoints'
 import { getFromEndpoint } from '../services/axios'
-import { formatDiscipline } from './helpers'
+import { formatDiscipline, isTypeLoadProtected, ServerResponseDiscipline } from './helpers'
 import useLoadAsync, { Return } from './useLoadAsync'
 
-export interface ServerResponse {
-  id: number
-  title: string
-  description: string
-  icon: string
-  total_training_sets: number
-  total_discipline_children: number
-  total_documents: number
-  nested_training_sets: number[]
-}
+export type DisciplinesRequestData =
+  | {
+      apiKey: string
+    }
+  | {
+      parent: Discipline | null
+    }
 
-const getEndpoint = (parent: Discipline | null): string => {
-  if (parent?.needsTrainingSetEndpoint) {
-    return ENDPOINTS.trainingSet
+const getEndpoint = (params: DisciplinesRequestData): string => {
+  if (isTypeLoadProtected(params)) {
+    return ENDPOINTS.groupInfo
+  }
+  if (params.parent?.needsTrainingSetEndpoint) {
```

Comment:
not all disciplines are get by the /disciplines/{parentId} endpoint. If they are leaf disciplines they need to get by the endpoint /training_set/{parentId}. This is just because of design of the cms endpoints.